mapper/orm/src/main/java/org/hibernate/search/mapper/orm/schema/management/SchemaManagementStrategyName.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.orm.schema.management;
+
+import java.lang.invoke.MethodHandles;
+
+import org.hibernate.search.mapper.orm.logging.impl.Log;
+import org.hibernate.search.util.common.impl.StringHelper;
+import org.hibernate.search.util.common.logging.impl.LoggerFactory;
+
+public enum SchemaManagementStrategyName {
+
+	/**
+	 * A strategy that does not do anything on startup or shutdown.
+	 * <p>
+	 * Indexes and their schema will not be created nor deleted on startup or shutdown.
+	 * Hibernate Search will not even check that the index actually exists.
+	 * <p>
+	 * With Elasticsearch, indexes and their schema will have to be created explicitly.
+	 */
+	NONE( "none" ),
+
+	/**
+	 * A strategy that does not change indexes nor their schema,
+	 * but checks that indexes exist and validates their schema on startup.
+	 * <p>
+	 * An exception will be thrown on startup if:
+	 * <ul>
+	 *     <li>Indexes are missing</li>
+	 *     <li>OR, with Elasticsearch only, indexes exist but their schema does not match the requirements
+	 *     of the Hibernate Search mapping:
+	 *     missing fields, fields with incorrect type, missing analyzer definitions or normalizer definitions, ...
+	 *     </li>
+	 * </ul>
+	 * <p>
+	 * With Lucene, the schema of indexes is not validated.
+	 */
+	VALIDATE( "validate" ),
+
+	/**
+	 * A strategy that creates missing indexes and their schema on startup,
+	 * but does not touch existing indexes and assumes their schema is correct without validating it.
+	 * <p>
+	 * Note that creating indexes and their schema will not populate the indexed data:
+	 * a newly created index will always be empty.
+	 * To populate indexes with pre-existing data, use mass indexing.
+	 */
+	CREATE( "create" ),
+
+	/**
+	 * A strategy that creates missing indexes and their schema on startup,
+	 * and validates the schema of existing indexes.
+	 * <p>
+	 * Note that creating indexes and their schema will not populate the indexed data:
+	 * newly created indexes will always be empty.
+	 * To populate indexes with pre-existing data, use mass indexing.
+	 * <p>
+	 * With Elasticsearch only, an exception will be thrown on startup if some indexes already exist
+	 * but their schema does not match the requirements of the Hibernate Search mapping:
+	 * missing fields, fields with incorrect type, missing analyzer definitions or normalizer definitions, ...
+	 * <p>
+	 * With Lucene, the schema of indexes is not validated.
+	 */
+	CREATE_OR_VALIDATE( "create-or-validate" ),
```

Comment:
Maybe it's not 100% clear that with Lucene any pre-existing schema will provoke an exception.

Reply:
It will, in fact, always pass as long as the index exists. I'll clarify.